tsl/src/continuous_aggs/repair.c

```diff
@@ -146,9 +139,6 @@ cagg_rebuild_view_definition(ContinuousAgg *agg, Hypertable *mat_ht, bool force_
 													&mataddress,
 													NameStr(mat_ht->fd.table_name));
 	}
-	else
-		view_query =
-			finalizequery_get_select_query(&fqi, mattblinfo.matcollist, &mataddress, relname);
```

Comment:
I am wondering what happens in this case: Since we now remove the else statement, `view_query` could be uninitialized in line 149. If `!agg->data.materialized_only` in line 153 is also false, the variable stays uninitialized and we will access the `NULL` pointer in line 181. 

Even if the condition is true, passing the uninitialized `view_query`  to `build_union_query` looks suspicious. However, our test cases do not fail and these branches are marked as dead code. But could you double-check that this is not a problem before merging?

Reply:
It will not happen because in the current logic the only way to this happen is if the `finalized=false` and in that case we are exiting earlier of the function emitting a warning message.

Reply:
Anyway removed the branch because it become useless at this point.